tests/foreman/cli/test_host.py

```diff
@@ -70,1769 +64,1491 @@
+    location = Location.info({'name': function_make_host['location']})
+    user = make_user(
+        {
+            'admin': False,
+            'default-organization-id': org['id'],
+            'organization-ids': [org['id']],
+            'default-location-id': location['id'],
+            'location-ids': [location['id']],
+            'login': user_name,
+            'password': user_password,
+        }
+    )
+    return {'user': user, 'password': user_password}
+
+
+# -------------------------- CREATE SCENARIOS -------------------------
+@pytest.mark.host_create
+@pytest.mark.tier1
+@pytest.mark.upgrade
+def test_positive_create_and_delete(module_lce_library, module_published_cv):
+    """A host can be created and deleted
+
+    :id: 59fcbe50-9c6b-4c3c-87b3-272b4b584fb3
+
+    :expectedresults: A host is created and deleted
+
+    :BZ: 1260697, 1313056, 1361309
+
+    :CaseImportance: Critical
+    """
+    name = valid_hosts_list()[0]
+    host = entities.Host()
+    host.create_missing()
+    interface = (
+        'type=interface,mac={},identifier=eth0,name={},domain_id={},'
+        'ip={},primary=true,provision=true'
+    ).format(host.mac, gen_string('alpha'), host.domain.id, gen_ipaddr())
+    new_host = make_host(
+        {
+            'architecture-id': host.architecture.id,
+            'content-view-id': module_published_cv.id,
+            'domain-id': host.domain.id,
+            'environment-id': host.environment.id,
+            'interface': interface,
+            'lifecycle-environment-id': module_lce_library.id,
+            'location-id': host.location.id,
+            'mac': host.mac,
+            'medium-id': host.medium.id,
+            'name': name,
+            'operatingsystem-id': host.operatingsystem.id,
+            'organization-id': host.organization.id,
+            'partition-table-id': host.ptable.id,
+            'root-password': host.root_pass,
+        }
+    )
+    assert f'{name}.{host.domain.read().name}' == new_host['name']
+    assert new_host['organization'] == host.organization.name
+    assert new_host['content-information']['content-view']['name'] == module_published_cv.name
+    assert (
+        new_host['content-information']['lifecycle-environment']['name'] == module_lce_library.name
+    )
+    host_interface = HostInterface.info(
+        {'host-id': new_host['id'], 'id': new_host['network-interfaces'][0]['id']}
+    )
+    assert host_interface['domain'] == host.domain.read().name
+
+    Host.delete({'id': new_host['id']})
+    with pytest.raises(CLIReturnCodeError):
+        Host.info({'id': new_host['id']})
+
+
+@pytest.mark.host_create
+@pytest.mark.tier1
+def test_positive_add_interface_by_id():
+    """New network interface can be added to existing host
+
+    :id: e97dba92-61eb-47ad-a7d7-5f989292b12a
+
+    :expectedresults: Interface added to host correctly and has proper
+        domain and mac address
+
+    :CaseImportance: Critical
+    """
+    domain = make_domain(
```

Comment:
this could be a fixture (perhaps there's even a global one already)

Reply:
this is question for discussion, basically most of the cli make_ functions can be replaced by api fixtures

I probably should replace everything possible.